air-script/tests/aux_trace/aux_trace.masm

```diff
@@ -1,57 +1,189 @@
+            dup.0 push.1966080 u32checked_and neq.0 # Included(20)..Included(17)
+            if.true
+                dup.0 push.393216 u32checked_and neq.0 # Included(18)..Included(17)
+                if.true
+                    push.131072 u32checked_and neq.0 # Test if trace length is a power of 2^17
+                    push.14996013474702747840 push.15139302138664925958 dup.2 cdrop push.15139302138664925958 push.9071788333329385449 movup.3 cdrop
+                else
+                    push.524288 u32checked_and neq.0 # Test if trace length is a power of 2^19
+                    push.6451340039662992847 push.5708508531096855759 dup.2 cdrop push.5708508531096855759 push.14996013474702747840 movup.3 cdrop
+                end # END else
+            else
+                dup.0 push.6291456 u32checked_and neq.0 # Included(22)..Included(21)
+                if.true
+                    push.2097152 u32checked_and neq.0 # Test if trace length is a power of 2^21
+                    push.10420286214021487819 push.5102364342718059185 dup.2 cdrop push.5102364342718059185 push.6451340039662992847 movup.3 cdrop
+                else
+                    push.8388608 u32checked_and neq.0 # Test if trace length is a power of 2^23
+                    push.17538441494603169704 push.13945510089405579673 dup.2 cdrop push.13945510089405579673 push.10420286214021487819 movup.3 cdrop
+                end # END else
+            end # END else
+        else
+            dup.0 push.503316480 u32checked_and neq.0 # Included(28)..Included(25)
+            if.true
+                dup.0 push.100663296 u32checked_and neq.0 # Included(26)..Included(25)
+                if.true
+                    push.33554432 u32checked_and neq.0 # Test if trace length is a power of 2^25
+                    push.8974194941257008806 push.16784649996768716373 dup.2 cdrop push.16784649996768716373 push.17538441494603169704 movup.3 cdrop
+                else
+                    push.134217728 u32checked_and neq.0 # Test if trace length is a power of 2^27
+                    push.5506647088734794298 push.16194875529212099076 dup.2 cdrop push.16194875529212099076 push.8974194941257008806 movup.3 cdrop
+                end # END else
+            else
+                dup.0 push.1610612736 u32checked_and neq.0 # Included(30)..Included(29)
+                if.true
+                    push.536870912 u32checked_and neq.0 # Test if trace length is a power of 2^29
+                    push.16558868196663692994 push.7731871677141058814 dup.2 cdrop push.7731871677141058814 push.5506647088734794298 movup.3 cdrop
+                else
+                    drop push.9896756522253134970 push.16558868196663692994
+                end # END else
+            end # END else
+        end # END else
+    end # END else
+end # END PROC get_exemptions_points
+
+proc.cache_z_exp
+    padw mem_loadw.4294903304 drop drop
+    # => [z_1, z_0, ...]
+    # Exponentiate z trace_len times
```

Comment:
The point $z^{tracelen-exemptions}$ for the boundary constraint divisor will be computed by dividing this value twice. This is because this procedure only performs a power-of-two exponentiation (useful to reuse results when period columns are used).